lib/ffi-libarchive/reader.rb

```diff
@@ -171,5 +185,57 @@ def save_data(file_name)
         raise Error, @archive if C.archive_read_data_into_fd(archive, fd) != C::OK
       end
     end
+
+    private
+
+    #
+    # See:
+    #   1. https://github.com/libarchive/libarchive/blob/6a9dcf9fc429e2dc9fb08e669bf7b0bed4d5edf9/tar/read.c#L346
+    #   2. https://github.com/libarchive/libarchive/blob/a11f15860ae39ecdc8173243a211cdafc8ac893c/tar/util.c#L523-L535
+    #   3. https://github.com/libarchive/libarchive/blob/a11f15860ae39ecdc8173243a211cdafc8ac893c/tar/util.c#L554-L560
```

Comment:
now i'm a bit confused as to why we're not just calling that code in libarchive directly?

Reply:
Because this isn't a part of libarchive (the library), it's a part of `tar` itself. Both are housed in the same git repository. I don't think tar is included in libarchive itself in distro's, etc.

Reply:
okay looking closer at the libarchive git repo that makes sense we'd have to copy code from an application that uses libarchive.